Update learn-accounts.md (#2358)
w3f/polkadot-wiki: docs/learn-accounts.md

Most wallets generate a mnemonic phrase for users to back up their wallets and generate a private
key from the mnemonic. Not all wallets use the same algorithm to convert from mnemonic phrase to

> private key. A typical mnemonic seed phrase generated by [the Subkey tool](https://substrate.dev/docs/en/knowledgebase/integrate/subkey) is shown below

Reviewer: ```suggestion
private key. A typical mnemonic phrase generated by [the Subkey tool](https://substrate.dev/docs/en/knowledgebase/integrate/subkey) is shown below
```

```js
"caution juice atom organ advance problem want pledge someone senior holiday very"
```

and its corresponding private/public key pair is shown below.

```
Secret seed (Private key): 0x056a6a4e203766ffbea3146967ef25e9daf677b14dc6f6ed8919b1983c9bebbc
Public key (SS58): 5F3sa2TJAWMqDhXG6jhV4N8ko9SxwGy8TpaNS1repo5EYjQX
```

Subkey and Polkadot-JS based wallets use the BIP39 dictionary for mnemonic generation, but use the
entropy byte array to generate the private key, while full BIP39 wallets (like Ledger) use 2048
rounds of PBKDF2 on the mnemonic. As such, the same mnemonic may generate different private keys on different wallets, due to the difference in the cryptographic algorithms used. See [Substrate BIP39](https://github.com/paritytech/substrate-bip39) for more information.

### Derivation Paths

> If you would like to create and manage several accounts on the network using the same mnemonic seed phrase, you can use derivation paths. We can think of the derived accounts as the child accounts of the root account created using the original mnemonic seed phrase. Many Polkadot key generation tools support hard and soft derivation. For instance, if you intend to create an account to be used on the Polkadot chain, you can derive a **hard key** child account using **//** after the mnemonic phrase.

Reviewer: ```suggestion
If you would like to create and manage several accounts on the network using the same seed, you can use derivation paths. We can think of the derived accounts as child accounts of the root account created using the original mnemonic seed phrase. Many Polkadot key generation tools support hard and soft derivation. For instance, if you intend to create an account to be used on the Polkadot chain, you can derive a **hard key** child account using **//** after the mnemonic phrase.
```

```js

> "caution juice atom organ advance problem want pledge someone senior holiday very//polkadot//0"

Reviewer: ```suggestion
"caution juice atom organ advance problem want pledge someone senior holiday very//0"
```
The "polkadot" isn't necessary, but is added by default by Parity Signer and probably some other wallets.

```

and a **soft key** child account using **/** after the mnemonic phrase

```js

> "caution juice atom organ advance problem want pledge someone senior holiday very/polkadot/0"

Reviewer: ```suggestion
"caution juice atom organ advance problem want pledge someone senior holiday very/0"
```

```

> If you would like to create another account for usage on the Polkadot chain, you can change the number at the end of the string above from 0 to 1. You can use any words or numbers in the derivation path as long as they make sense to you. A soft derivation allows you to "go backwards" to figure out previous private keys IF you know the second private key, and it's possible to determine that different accounts generated from the same seed are linked to that seed. A hard derivation path does not allow either of these - if you know a derived private key, it's not feasible to figure out the initial private key of the root address, and it's impossible to prove that the first account is linked with the second. Both of these derivation methods have their use cases, given the private keys for all the derived accounts are fully secure. If it is likely that you may use a derived account in a less secure setting, it is recommended to generate the account using a hard derivation path.

Reviewer: ```suggestion
If you would like to create another account for usage on the Polkadot chain using the same seed, you can change the number at the end of the string above. For example, `/1`, `/2`, and `/3` will all create different derived accounts.

You can use any letters or numbers in the derivation path as long as they make sense to you; they do not have to follow any specific pattern. You may also combine multiple derivations in your path, as well. For instance, `//bill//account//1` and `//john/polkadot/initial` are both valid. You will need to know both the seed and the derivation path in order to recreate a derived account, so you should either use a well-defined sequence (e.g. //0, //1, //2...) or be sure to write down any derivation paths you use.

> It is not possible to generate a derived account without also knowing the derivation path.

### Soft vs. Hard Derivation

A soft derivation allows someone to potentially "go backwards" to figure out the private key of the initial account IF they know the private key of the derived account. It is also possible to determine that different accounts generated from the same seed are linked to that seed. A hard derivation path does not allow either of these - even if you know a derived private key, it's not feasible to figure out the private key of the root address, and it's impossible to prove that the first account is linked with the second. Both of these derivation methods have their use cases, given the private keys for all the derived accounts are fully secure. Unless you have a specific need for a soft derivation, it is recommended to generate the account using a hard derivation path.
```

Reviewer: We should also put the Soft vs. Hard description further up above (before we mention using / and //)
